src/devices/Board/Board.cs

```diff
@@ -0,0 +1,407 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Iot.Device.Board
+{
+    public abstract class Board : MarshalByRefObject, IDisposable
+    {
+        private readonly PinNumberingScheme _defaultNumberingScheme;
+        private readonly object _pinReservationsLock;
+        private readonly Dictionary<int, List<PinReservation>> _pinReservations;
+        private bool _initialized;
+        private bool _disposed;
+
+        protected Board(PinNumberingScheme defaultNumberingScheme)
+        {
+            _defaultNumberingScheme = defaultNumberingScheme;
+            _pinReservations = new Dictionary<int, List<PinReservation>>();
+            _pinReservationsLock = new object();
+            _initialized = false;
+            _disposed = false;
+        }
+
+        ~Board()
+        {
+            Dispose(false);
+        }
+
+        public event Action<string, Exception> LogMessages;
+
+        protected bool Initialized
+        {
+            get
+            {
+                return _initialized;
+            }
+        }
+
+        protected bool Disposed
+        {
+            get
+            {
+                return _disposed;
+            }
+        }
+
+        public PinNumberingScheme DefaultPinNumberingScheme
+        {
+            get
+            {
+                return _defaultNumberingScheme;
+            }
+        }
+
+        protected void Log(string message, Exception exception = null)
+        {
+            LogMessages?.Invoke(message, null);
```

Comment:
I'd prefer logging wasn't part of Board abstraction, it can use it but it should not implement or abstract it

Reply:
I think I'll put up a small generic logging API, to be usable for all bindings.

Reply:
... but on a separate PR.

Reply:
Can we reuse the logger I've been using for the nfc readers? 
https://github.com/dotnet/iot/blob/master/src/devices/Card/LogInfo.cs 

Should be generic enough. We can move to the common folder maybe?

Reply:
I'll check what we already have, yes. Just this shouldn't be part of this PR (even if it means that these methods will change again)

Reply:
I left this in for now, but plan to replace it once we agree on a logging system/library.

Reply:
let's do it separately regardless of decision

Reply:
Removed, it's not used in this class so far, anyway.